### PR TITLE
join: remove a clippy::cognitive_complexity by moving some content in functions

### PR DESCRIPTION
**Pull request by sylvestre**

### CODE REVIEW

File: src/uu/join/src/join.rs [hunk 1]
```diff
@@ -591,11 +591,11 @@ impl<'a> State<'a> {
...
-#[uucore::main]
-#[allow(clippy::cognitive_complexity)]
-pub fn uumain(args: impl uucore::Args) -> UResult<()> {
-    let matches = uu_app().try_get_matches_from(args)?;
-
+/// Parses the command-line arguments and constructs a `Settings` struct.
+///
+/// This function takes the matches from the command-line arguments, processes them,
+/// and returns a `Settings` struct that encapsulates the configuration for the program.
+fn parse_settings(matches: &clap::ArgMatches) -> UResult<Settings> {
```
Comment (cakebaker) [inline src/uu/join/src/join.rs:598]: Unfortunately, the refactorings are not enough and the `cognitive_complexity` lint still shows a warning.
```suggestion
#[allow(clippy::cognitive_complexity)]
fn parse_settings(matches: &clap::ArgMatches) -> UResult<Settings> {
```

File: src/uu/join/src/join.rs [hunk 2]
```diff
@@ -677,12 +677,29 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
...
 
+/// Validates the provided file inputs.
+///
+/// This function checks if both files are standard input and returns an error if they are.
+fn validate_files(file1: &str, file2: &str) -> UResult<()> {
     if file1 == "-" && file2 == "-" {
         return Err(USimpleError::new(1, "both files cannot be standard input"));
     }
+    Ok(())
+}
```
Comment (cakebaker) [inline src/uu/join/src/join.rs:691]: I think it is overkill to have a function for this trivial functionality as it is no shared functionality.

File: src/uu/join/src/join.rs (hunk 1 above)
Comment (sylvestre) [reply src/uu/join/src/join.rs:598]: oh, maybe we should break the clippy task on this too

Comment (cakebaker) [reply src/uu/join/src/join.rs:622]: Hm, what do you mean with "break the clippy task"? Currently, the lint still shows a warning.

Comment (sylvestre) [reply src/uu/join/src/join.rs:622]: like this:
https://github.com/uutils/coreutils/pull/5379

Comment (cakebaker) [reply src/uu/join/src/join.rs:622]: Sorry, I still don't understand what you mean :| #5379 has been merged in the meantime and it shows an error for this PR (for some reason the job doesn't fail):
```
error: the function has a cognitive complexity of (12/10)
   --> src/uu/join/src/join.rs:622:4
    |
622 | fn parse_settings(matches: &clap::ArgMatches) -> UResult<Settings> {
```

File: src/uu/join/src/join.rs [hunk 3]
```diff
@@ -591,62 +591,81 @@ impl<'a> State<'a> {
...
+fn get_and_parse_field_number(matches: &clap::ArgMatches, key: &str) -> UResult<Option<usize>> {
+    let value = matches.get_one::<String>(key).map(|s| s.as_str());
+    parse_field_number_option(value)
+}
+
+/// Parses the command-line arguments and constructs a `Settings` struct.
+///
+/// This function takes the matches from the command-line arguments, processes them,
+/// and returns a `Settings` struct that encapsulates the configuration for the program.
+fn parse_settings(matches: &clap::ArgMatches) -> UResult<Settings> {
```
Comment (cakebaker) [inline src/uu/join/src/join.rs:650]:
```suggestion
#[allow(clippy::field_reassign_with_default)]
fn parse_settings(matches: &clap::ArgMatches) -> UResult<Settings> {
```

**Review by cakebaker — APPROVED** [review]

**Pull request merged**